fabric-renderer-indigo/src/client/java/net/fabricmc/fabric/impl/client/indigo/renderer/helper/ColorHelper.java

```diff
@@ -28,13 +26,13 @@
 public abstract class ColorHelper {
 	private ColorHelper() { }
 
-	private static final Int2IntFunction colorSwapper = ByteOrder.nativeOrder() == ByteOrder.LITTLE_ENDIAN ? color -> ((color & 0xFF00FF00) | ((color & 0x00FF0000) >> 16) | ((color & 0xFF) << 16)) : color -> color;
```

Comment:
These constant lambdas are very fast, likely faster than an `if` check every time, and they should get inlined by hotspot. I suspect it was done this way for performance reasons in the first place.

Reply:
FREX base switched to a static final boolean as well. There is a comment saying that if checks with static final booleans are inlined, which if true, would be faster than using a function. I had not ever heard that functions like this can be inlined. I will ask a few community members that know more about how the JIT works.

Reply:
Both get inlined, the resulting assembly is identical: https://gist.github.com/LlamaLad7/0c4c39b1e86f4ebb8bd01bd4c6f9f2d2.
The JIT is very good at its job and there is normally no need to make such micro-optimizations manually.

Reply:
Ah, that's great. Thank you Llama. In that case we should probably keep the boolean check in `ColorHelper` to keep the quad emitting code more readable.

Reply:
The red blue color swap has been removed completely in the latest commit.